plugins/sentinelone/komand_sentinelone/tasks/monitor_activities_and_events/task.py

```diff
@@ -0,0 +1,327 @@
+import insightconnect_plugin_runtime
+import requests
+from insightconnect_plugin_runtime.exceptions import HTTPStatusCodes, PluginException
+from insightconnect_plugin_runtime.helper import extract_json, response_handler
+from .schema import (
+    MonitorActivitiesAndEventsInput,
+    MonitorActivitiesAndEventsOutput,
+    MonitorActivitiesAndEventsState,
+    Input,
+    Output,
+    Component,
+    State,
+)
+from komand_sentinelone.util.exceptions import ApiException
+
+# Custom imports below
+from datetime import datetime, timezone, timedelta
+from typing import List, Dict, Tuple, Optional
+
+# State values
+LAST_RUN_TIMESTAMP = "last_run_timestamp"
+ACTIVITIES_LAST_LOG_TIMESTAMP = "activities_last_log_timestamp"
+EVENTS_LAST_LOG_TIMESTAMP = "events_last_log_timestamp"
+THREATS_LAST_LOG_TIMESTAMP = "threats_last_log_timestamp"
+ACTIVITIES_PAGE_CURSOR = "activities_page_cursor"
+EVENTS_PAGE_CURSOR = "events_page_cursor"
+THREATS_PAGE_CURSOR = "threats_page_cursor"
+
+# Cutoff values
+DATE_TIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+INITIAL_CUTOFF_HOURS = 24
+MAX_CUTOFF_HOURS = 24 * 7
+LOGS_PAGE_LIMIT = 1000
+
+API_VERSION = "2.1"
+STARTING_TIMESTAMP = "1900-01-01T00:00:00.000000Z"
+
+ACTIVITIES_LOGS = "activities_logs"
+EVENTS_LOGS = "events_logs"
+THREATS_LOGS = "threats_logs"
+
+LOG_TYPE_MAP = {
+    ACTIVITIES_LOGS: {"LAST_LOG_TIMESTAMP": ACTIVITIES_LAST_LOG_TIMESTAMP, "PAGE_TOKEN": ACTIVITIES_PAGE_CURSOR},
+    EVENTS_LOGS: {"LAST_LOG_TIMESTAMP": EVENTS_LAST_LOG_TIMESTAMP, "PAGE_TOKEN": EVENTS_PAGE_CURSOR},
+    THREATS_LOGS: {"LAST_LOG_TIMESTAMP": THREATS_LAST_LOG_TIMESTAMP, "PAGE_TOKEN": THREATS_PAGE_CURSOR},
+}
+
+
+class MonitorActivitiesAndEvents(insightconnect_plugin_runtime.Task):
```

Comment:
Suggestion: Rename to something more generic like MonitorLogs as this seems to retrieve threats as well as activities and events?